test/machi_file_proxy_eqc.erl

```diff
@@ -130,6 +136,80 @@ check_writes(write, [{Pos, Sz}|_T], Off, L) when ( Off + L ) > Pos
 check_writes(Op, [_H|T], Off, L) ->
     check_writes(Op, T, Off, L).
 
+get_overlaps(_Offset, _Len, [], Acc) -> lists:reverse(Acc);
+get_overlaps(Offset, Len, [{Pos, Sz} = Ck|T], Acc0)
+%% Overlap judgement differnt from the one in machi_csum_table
+%% [a=Offset, b), [x=Pos, y) ...
+  when
+      %% (a-y) * (b-x)
+      %%(Offset - Pos - Sz) * (Offset + Len - Pos) < 0 ->
+      %% a x b y
+      (Offset =< Pos andalso Pos < Offset + Len andalso Offset + Len =< Pos + Sz) orelse
+      %% a x y b
+      (Offset =< Pos andalso Pos + Sz < Offset + Len) orelse
```

Comment:
Ah, I may be too quick and injudicious. The boundary treatment is for mutual exclusion between four conditions? If so, comments like `a <= x <= b <= y` and `a <= x < (<=?) y < b` and... may help ?